File: Code/Components/DanpheEMR.ServerModel/WardSupplyModels/WARDInventoryConsumptionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class WARDInventoryConsumptionModel
    {
        [Key]
        public int ConsumptionId { get; set; }
        public int StoreId { get; set; }
        public int DepartmentId { get; set; }
        public string DepartmentName { get; set; }
        public int ItemId { get; set; }
        public string ItemName { get; set; }
        public double Quantity { get; set; }
        public string Remark { get; set; }
        public string UsedBy { get; set; }
        public int CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ConsumptionDate { get; set; }
        [NotMapped]
        public int CounterId { get; set; }
        [NotMapped]
        public double ConsumeQuantity { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/wardsupply/shared/ward-inventory-consumption.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms'
import { ItemMaster } from '../../inventory/shared/item-master.model';
export class WardInventoryConsumptionModel {
  public ConsumptionId: number = 0;
  public DepartmentId: number = 0;
  public DepartmentName: string = null;
  public ItemId: number = 0;
  public ItemName: string = '';
  public Quantity: number = 0;
  public Code: string = '';
  public UOMName: string = '';
  public Remark: string = '';
  public UsedBy: string = null;
  public CreatedBy: number = 0;
  public CreatedOn: string = '';
  public ConsumptionDate: string = '';
  public CounterId: number = 0;
  public StockId: number = 0;
  public StoreId: number = 0;
  public ConsumeQuantity: number = 1;
  public SelectedItem: ItemMaster = null;

  public ConsumptionValidator: FormGroup = null;
  constructor() {
    var _formBuilder = new FormBuilder();
    this.ConsumptionValidator = _formBuilder.group({
      'ConsumeQuantity': ['', Validators.compose([Validators.required,Validators.min(1)])]
    });
  }
  public IsDirty(fieldName): boolean {
    if (fieldName == undefined)
      return this.ConsumptionValidator.dirty;
    else
      return this.ConsumptionValidator.controls[fieldName].dirty;
  }


  public IsValid(fieldName, validator): boolean {
    if (fieldName == undefined) {
      return this.ConsumptionValidator.valid;
    }
    else
      return !(this.ConsumptionValidator.hasError(validator, fieldName));
  }
}
